src/rust/ide/src/ide/integration.rs

```diff
@@ -324,19 +324,19 @@ impl Integration {
         }
 
         frp::extend! { network
-            eval_ editor_outs.node_editing_started([]{analytics::remote_log(analytics::AnonymousData("graph_editor::node_editing_started"))});
-            eval_ editor_outs.node_editing_finished([]{analytics::remote_log(analytics::AnonymousData("graph_editor::node_editing_finished"))});
-            eval_ editor_outs.node_added([]{analytics::remote_log(analytics::AnonymousData("graph_editor::node_added"))});
-            eval_ editor_outs.node_removed([]{analytics::remote_log(analytics::AnonymousData("graph_editor::node_removed"))});
-            eval_ editor_outs.nodes_collapsed([]{analytics::remote_log(analytics::AnonymousData("graph_editor::nodes_collapsed"))});
-            eval_ editor_outs.node_entered([]{analytics::remote_log(analytics::AnonymousData("graph_editor::node_enter_request"))});
-            eval_ editor_outs.node_exited([]{analytics::remote_log(analytics::AnonymousData("graph_editor::node_exit_request"))});
-            eval_ editor_outs.on_edge_endpoints_set([]{analytics::remote_log(analytics::AnonymousData("graph_editor::edge_endpoints_set"))});
-            eval_ editor_outs.visualization_enabled([]{analytics::remote_log(analytics::AnonymousData("graph_editor::visualization_enabled"))});
-            eval_ editor_outs.visualization_disabled([]{analytics::remote_log(analytics::AnonymousData("graph_editor::visualization_disabled"))});
-            eval_ on_connection_removed([]{analytics::remote_log(analytics::AnonymousData("graph_editor::connection_removed"))});
-            eval_ searcher_frp.used_as_suggestion([]{analytics::remote_log(analytics::AnonymousData("searcher::used_as_suggestion"))});
-            eval_ project_frp.editing_committed([]{analytics::remote_log(analytics::AnonymousData("project::editing_committed"))});
+            eval_ editor_outs.node_editing_started([]{analytics::remote_log_event("graph_editor::node_editing_started")});
```

Comment:
What hapenned with all the `AnonymousData` things here? We were designing them for a purpose. Why they are deleted? Also, why the name change from `remote_log` to `remote_log_event`?

Reply:
The message title is not meant to be `data` but a simple static event name in the source code. So, it cannot contain sensitive data and does not need to be wrapped.

`remote_log_event` is used to indicate that only the presence of an event is logged, while data needs to be logged with `remote_log_value`

Reply:
I don't "feel" it tbh. I mean, I understand the purpose, but in such a scenario I think we can remove `AnonymousData` altogether. IMO either we keep it secure by writing in docs it should be secure (like in case of `remote_log_event`), or we use some newtype machinery in code to be sure everybody remembers about it. I can think of theoretical event names that violate the anonymous data collection rules. I know this example is not perfect, but `"node_created_by_a_woman"` would violate it. And of course, we would catch it during code review, but we would probably catch it as well during code review in case of `remote_log_value`. I think that we should either use `AnonymousData` everywhere in the API or nowhere, but I'm of course very OK with chatting ore about it! CC @farmaazon , I'd love to hear your thoughts about it too.

Reply:
I was considering to require a `&'static str` which almost enforces the use of a string literal here. How about we create an issue for the discussion about proper refactoring and handling of this?